src/generic/stage2/numberparsing.h

```diff
@@ -460,7 +460,8 @@ really_inline bool parse_number(const uint8_t *const src, W &writer) {
     if (!parse_exponent(src, p, exponent)) { return false; }
   }
   if (is_float) {
-    return write_float(src, negative, i, start_digits, digit_count, exponent, writer);
+    const bool clean_end = is_structural_or_whitespace(*p);
```

Comment:
I like your code proposal because it is more uniform with what we already have.

Reply:
Nah. It results in worse assembly under GNU GCC9.

With your code proposal...

|numbers.json|0.911|0.899|-1.3%|
|marine_ik.json|0.830|0.809|-2.5%|
|mesh.json|0.818|0.797|-2.5%|
|canada.json|0.794|0.771|-2.9%|
|mesh.pretty.json|1.336|1.284|-3.9%|

current...

|numbers.json|0.909|0.904|-0.5%|
|marine_ik.json|0.829|0.818|-1.3%|
|mesh.json|0.819|0.801|-2.2%|
|canada.json|0.794|0.782|-1.5%|
|mesh.pretty.json|1.339|1.316|-1.7%|

This might be very much compiler specific, of course, but if I can minimize the hit.